tools/autograd/derivatives.yaml

```diff
@@ -1894,6 +1894,15 @@
   self: max_unpool3d_backward(grad, self, indices, output_size, stride, padding)
   indices: non_differentiable
 
+- name: convolution(Tensor input, Tensor weight, Tensor? bias, int[] stride, int[] padding, int[] dilation, bool transposed, int[] output_padding, int groups) -> Tensor
+  input, weight, bias: "grad.defined() ? convolution_backward(grad, input, weight, bias->sizes(), stride, padding, dilation, transposed, output_padding, groups, grad_input_mask) : std::tuple<Tensor, Tensor, Tensor>()"
```

Comment:
nit: Maybe it would be nicer to move the grad.defined() check inside convolution_backward ?
Or there is a reason why this is not possible?

Reply:
I don't think it's possible - trying to do this gives me an error in the generated autograd code before the call to `convolution_backward` happens.

Reply:
Ok, non-CompositeImplicit function cannot take undefined Tensors as input.  LGTM then